src/FileServiceManager/FileService/DAL/Interfaces/IStorage.cs

```diff
@@ -0,0 +1,21 @@
+﻿using FileService.Models;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace FileService.DAL
+{
+    public interface IStorage
+    {
+        Task<Guid> UploadFileAsync(IFormFile file, int companyId);
+
+        Task<IEnumerable<FileMetadata>> GetAllFilesAsync(int companyId);
+
+        Task<FileMetadata> DownLoadFileAsync(Guid fileId, int companyId);
```

Comment:
download should open browsers download experience, please consider changing this method signature and the implementation in future appropriately

Reply:
I've changed to FileResult as per our conversation